homeassistant/components/nest.py

```diff
@@ -51,23 +55,44 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
-def request_configuration(nest, hass, config):
+async def async_nest_update_event_broker(hass, nest):
+    """
+    Fire a EVENT_NEST_UPDATE event when nest stream API received data.
+
+    nest.update_event.wait will block the thread in most of time,
+    so specific an executor to save default thread pool.
+    """
+    _LOGGER.debug("listening nest.update_event")
+    with ThreadPoolExecutor(max_workers=1) as executor:
```

Comment:
I don't think we should do this. Can't python-nest allow a callback to be registered? The callback should be called by python-nest when a state update is needed.

Otherwise waiting for the event should be done by a coroutine asynchronously. But the current event is a `threading.Event` so I don't think that will work as it is now.

Reply:
My code works. `run_in_executor(event.wait)` is the way to glue `threading.Event` and `asyncio`, and `async_nest_update_event_broker` is a coroutine.

Reply:
Yes, but it's blocking a thread, which isn't good.

Reply:
That is why I am creating a new executor, so that I just block my own thread, won't affect others.

python-nest's author is not responsive like you guys, the stream API changes have to wait months to get review and merged. I think we better to use whatever we have now.

BTW, I really dig myself into a hole. That update_event was introduce by myself to the python-nest, I should allow caller pass in an optional event object

Reply:
The new executor is what I don't like.

Maybe @balloob has a better idea than me how to do this?

Reply:
The dev of python-nest seems active right now, if it's a small change should probably be able to get in quickly?

Reply:
What is the prefer way? An simple callback function enough?

Reply:
@balloob I am rethinking that under the influence of caffeine, I feel like current implement is not bad from python-nest perspective. As a service provider, it doesn't know what its client will put in the callback, it might be better just use thread event to loosen the coulping.

My next  change for python-nest will be allow only monitor part of the data, so I may need multiple update event, and I may try to async-ized whole package, at that time, I will provide an better interface for coroutine call back.

Reply:
Ok, that sounds good. If you go async the whole package, consider allowing passing in an aiohttp client session, that way it is best suited for HASS 😉